### PR TITLE
feat: disable workflows and webhooks for orgs

### LINKED ISSUE #9359: [CAL-1873] Workflows for teams in an Org
[Org → Workflows \[3.0\] - 5. Organizations (Figma)](https://www.figma.com/file/mWENDNaJPdRfZMtA75MghZ/5.-Organizations?type=design&node-id=423%3A75747&t=mmpaixpgqk6GTSam-1)

* Workflows can be created by org admins for any team within an org
* Team admins can create workflows for their teams
* Org admins and team admins can also create workflows for their own links
* Members cannot create any workflows.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1873](https://linear.app/calcom/issue/CAL-1873/workflows-for-teams-in-an-org)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

For now, we don't want to allow creating org-wide workflows and webhooks. Also, some UI changes are made

Fixes #9359

[Org → Workflows \[3.0\] - 5. Organizations (Figma)](https://www.figma.com/file/mWENDNaJPdRfZMtA75MghZ/5.-Organizations?type=design&node-id=423%3A75747&t=mmpaixpgqk6GTSam-1)

Workflows: 
Screenshot 2023-06-13 at 14 00 06

Screenshot 2023-06-13 at 15 59 40

Webhooks: 
Screenshot 2023-06-13 at 14 17 06

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/ee/workflows/components/WorkflowListPage.tsx [hunk 1]
```diff
@@ -96,9 +98,9 @@ export default function WorkflowListPage({ workflows, profileOptions, hasNoWorkf
                         </div>
                       </div>
 
-                      <ul className="mt-1 flex flex-wrap space-x-2 sm:flex-nowrap ">
+                      <ul className="mt-2 flex flex-wrap space-x-2 sm:flex-nowrap ">
                         <li>
-                          <Badge variant="gray">
+                          <Badge size="lg" variant="gray">
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/WorkflowListPage.tsx:103]: If we do that in workflows and routing forms (https://github.com/calcom/cal.com/pull/9417#issuecomment-1587033998) we should also do it in event-types

File: packages/trpc/server/routers/viewer/workflows/getByViewer.handler.ts [hunk 2]
```diff
@@ -94,20 +99,45 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
...
-      },
-      metadata: {
-        readOnly: membership.role === MembershipRole.MEMBER,
-      },
-      workflows: membership.team.workflows,
-    }))
+    user.teams
+      .filter((mmship) => {
+        const metadata = teamMetadataSchema.parse(mmship.team.metadata);
+        return !metadata?.isOrganization;
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/getByViewer.handler.ts:112]: filter out orgs (workflows)

File: packages/trpc/server/routers/viewer/workflows/getByViewer.handler.ts [hunk 3]
```diff
@@ -94,20 +99,45 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
     },
   });
 
+  const teamMemberships = user.teams.map((membership) => ({
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/getByViewer.handler.ts:102]: same here (changes of this file are pretty much the same as we have it in `getByViewer.handler` for event types)

File: packages/trpc/server/routers/viewer/workflows/getByViewer.handler.ts (hunk 2 above)
Comment (joeauyeung) [reply packages/trpc/server/routers/viewer/workflows/getByViewer.handler.ts:112]: If it's the same logic between workflows & webhooks, can we abstract this into it's own function?

**Review by joeauyeung — APPROVED** [review]
LGTM. Just left one non-blocking comment

Tested by:
- Creating an org and a child team
- Going to workflow page and creating a new workflow
    - Org was not an option
- Going to webhook page and creating a new webhook
    - Org was not an option

**Comment by CarinaWolli** [comment]
> If it's the same logic between workflows & webhooks, can we abstract this into it's own function?

@joeauyeung Yes and event types have a similar logic as well. Definitely worth looking into abstracting it as we had to make the changes in three places

**Comment by CarinaWolli** [comment]
I will open a new PR that goes to `hariom/cal-1854-routing-forms-orgs-support` (https://github.com/calcom/cal.com/pull/9417) because I am reusing the `TeamsFilter` from this branch

**Pull request closed without merging**